### ISSUE TITLE
UnboundLocalError in langchain.retrievers.google_vertex_ai_search 

### ISSUE THREAD

**Issue by louridas** [langchain-ai/langchain#12780]
### System Info

Version 0.0.328, macOS, Python 3.11.5

In [_convert_website_search_response() ](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/retrievers/google_vertex_ai_search.py#L120-L154) of [langchain.retrievers.google_vertex_ai_search](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/retrievers/google_vertex_ai_search.py), if no documents are found it will try and print:

```
print(
                f"No {chunk_type} could be found.\n"
                "Make sure that your data store is using Advanced Website Indexing.\n"
                "https://cloud.google.com/generative-ai-app-builder/docs/about-advanced-features#advanced-website-indexing"  # noqa: E501
            )
```

However, the `chunk_type` variable is defined only if documents have been found, and is therefore unbound, leading to exception and crash.

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Any query that will fail to produce documents at all.

### Expected behavior

The variable should be bound so that the exception is avoided.

**Comment by spazm** [comment]
@dosu-bot where and how is `google_vertex_ai_search.py` tested?  How would we add a new test for this behavior?

**Comment by devstein** [comment]
Hey @spazm great question for Dosu, but as of now Dosu only response to the issue author and maintainers. Stay tuned for changes to this in the coming months 🦾 

Let me know if you have any feedback or feature you'd like us Dosu to support!

**Comment by baskaryan** [comment]
sounds like this was fixed in #12995
